File: examples/node/apibara-client.js
import "dotenv/config";

const API_KEY = process.env.APIBARA_API_KEY;
const BASE_URL = process.env.APIBARA_BASE_URL || "https://apibara.tech/api/v1/vehicle-auction";

if (!API_KEY) {
  throw new Error("Missing APIBARA_API_KEY environment variable.");
}

export async function apibaraGet(path, params = {}) {
  const url = new URL(`${BASE_URL}${path}`);

  for (const [key, value] of Object.entries(params)) {
    if (value !== undefined && value !== null && value !== "") {
      url.searchParams.set(key, String(value));
    }
  }

  const response = await fetch(url, {
    headers: {
      "X-API-Key": API_KEY,
      "Accept": "application/json"
    }
  });

  const bodyText = await response.text();

  if (!response.ok) {
    throw new Error(`Apibara API error ${response.status}: ${bodyText}`);
  }

  return bodyText ? JSON.parse(bodyText) : null;
}

File: examples/python/apibara_client.py
import os
from typing import Any, Dict, Optional

import requests
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.getenv("APIBARA_API_KEY")
BASE_URL = os.getenv("APIBARA_BASE_URL", "https://apibara.tech/api/v1/vehicle-auction")

if not API_KEY:
    raise RuntimeError("Missing APIBARA_API_KEY environment variable.")


def apibara_get(path: str, params: Optional[Dict[str, Any]] = None) -> Any:
    response = requests.get(
        f"{BASE_URL}{path}",
        headers={
            "X-API-Key": API_KEY,
            "Accept": "application/json",
        },
        params={k: v for k, v in (params or {}).items() if v is not None and v != ""},
        timeout=30,
    )

    response.raise_for_status()
    return response.json()
